velox/type/DecimalUtil.h

```diff
@@ -147,6 +148,55 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  /// Rescale a double value to decimal value.
+  ///
+  /// Use `folly::tryTo` to convert a double value to int128_t or int64_t. It
+  /// returns an error when overflow occurs so that we could determine whether
+  /// an overflow occurs through checking the result.
+  ///
+  /// Normally, return the rescaled value. Otherwise, if the `toValue` overflows
+  /// the TOutput's limits or the `toValue` exceeds the precision's limits, it
+  /// will throw an exception.
+  template <typename TOutput>
+  inline static TOutput
+  rescaleDouble(double inputValue, const int toPrecision, const int toScale) {
+    VELOX_USER_CHECK(
+        std::isfinite(inputValue),
+        "Cannot cast {} '{}' to DECIMAL({},{})",
```

Comment:
This is not a cast function, so can we align the error message with the purpose?

Reply:
But this function is only used for casting, just like the function `rescaleInt`.